src/tests/lossy_link.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <mutex>
+#include <random>
+
+class LossyLink
+{
+public:
+    explicit LossyLink(int seed_num);
+    ~LossyLink() = default;
+
+    /* Checks if we should drop a message given a ratio of dropped messages.
+     *
+     * @param ratio Ratio of drops (0: no drops, 1: everything dropped)
+     * @return true if message should be dropped.
+     */
+    bool drop(float ratio);
+
+private:
+    std::mutex _mutex{};
+    // We need predictable results, so we always start with the same seed.
+    std::seed_seq _seed;
```

Comment:
This doesn't have to be stored.

Reply:
True, thx.

Reply:
I couldn't do it :confused: .

Reply:
What does that mean?

Reply:
You have to help me with C++ :confused:

Reply:
Here you go: https://github.com/Auterion/mavlink-testing-suite/pull/10/commits/6b566b9169cfe5e355e8e6f0d9dea4cfcdec59f7